src/lib.rs

```diff
@@ -19,7 +19,12 @@ println!("{:?}", result.triangles); // [0, 2, 1, 0, 3, 2]
 ```
 */
 
-use std::{f64, fmt};
+#![no_std]
+
+extern crate alloc;
```

Comment:
Thanks for adding no_std support!
Trying this out locally my compiler is unhappy about two things:

1. vec! macros -> seems that adding #[macro_use] on top of the alloc extern solves it
2. abs() -> seems like the f64::abs is a wrapper on std and there is a core function that can be used instead? https://github.com/rust-lang/rust/issues/34766

Reply:
1. Fixed
2. abs() is easy enough to emulate on `no_std`. I added the `f64_abs` method which should be the full equivalent of `.abs()`, up to and including `NaN` behavior.

Reply:
Thanks for the update. Are you getting any other errors? When I tried to run compile the changes and run the tests I hit a few other problems:
```
error[E0599]: no method named `sqrt` found for type `f64` in the current scope
   --> /home/andre/.cargo/git/checkouts/delaunator-rs-c6e35a2502bd8582/a78aadd/src/lib.rs:289:35
    |
289 |         let hash_len = (n as f64).sqrt() as usize;
    |                                   ^^^^ method not found in `f64`

error[E0599]: no method named `abs` found for type `f64` in the current scope
   --> /home/andre/.cargo/git/checkouts/delaunator-rs-c6e35a2502bd8582/a78aadd/src/lib.rs:322:26
    |
322 |         let p = dx / (dx.abs() + dy.abs());
    |                          ^^^ method not found in `f64`

error[E0599]: no method named `abs` found for type `f64` in the current scope
   --> /home/andre/.cargo/git/checkouts/delaunator-rs-c6e35a2502bd8582/a78aadd/src/lib.rs:322:37
    |
322 |         let p = dx / (dx.abs() + dy.abs());
    |                                     ^^^ method not found in `f64`

error[E0599]: no method named `floor` found for type `f64` in the current scope
   --> /home/andre/.cargo/git/checkouts/delaunator-rs-c6e35a2502bd8582/a78aadd/src/lib.rs:326:29
    |
326 |         (((len as f64) * a).floor() as usize) % len
    |                             ^^^^^ method not found in `f64`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0599`.
error: could not compile `delaunator`
```

An idea you may want to consider is to bring in https://docs.rs/ieee754/0.2.6/ieee754/ as an optional dependency and make no_std a feature flag that when enabled bring in that dependency.
I saw this is what [robust](https://github.com/georust/robust/blob/6b482661d48207ebb0ff1c4bdc74aad5c6fc0387/Cargo.toml#L17) does and they use that crate for all the float operations not in core, like [abs]((https://github.com/georust/robust/blob/6b482661d48207ebb0ff1c4bdc74aad5c6fc0387/src/lib.rs#L1175).

In the other PR we have I am planning to bring it in as a dependency for robustness checks.
That means if we go this route, we would not be adding an extra dependency for no_std in practice because we would have to bring it in indirectly anyways from the robust crate.

What do you think?

Reply:
Hmm, I didn't see `sqrt` and `floor` when I was looking through this crate. There's no real way to implement those on `no_std` accurately. [micromath](https://crates.io/crates/micromath) exists, but most of its operations come with unacceptable degrees of inaccuracy (e.g. the `sqrt` function has an inaccuracy of 5%, which might prevent this crate from giving accurate results).

Reply:
`sqrt` is only used for determining integer hash table length (a speedup where inaccuracies don't really matter), so we could replace it with a very rough estimation — this value doesn't impact the final result.

Reply:
Fair nuff, then. It's been done.

Reply:
Awesome! Thanks for the changes. Tests are passing on this crate and also on voronoice crate.